web/src/routes/admin/constraints/+page.svelte

```diff
@@ -0,0 +1,124 @@
+<script lang="ts">
+  import { goto } from '$app/navigation';
+  import ConstraintSignatureComp from '$lib/components/ui/constraintSignature/constraint-signature.svelte';
+  import { Button } from '$lib/elements/ui/button';
+  import type { ComboBoxItem } from '$lib/elements/ui/combo-box/combo-box.svelte';
+  import DataTable from '$lib/elements/ui/dataTable/data-table.svelte';
+  import {
+    type ConstraintArgumentRequestDto,
+    type ConstraintInstanceRequestDto,
+    type ConstraintSignature,
+    createConstraintInstance,
+    deleteConstraintInstance,
+    getConstraintInstances,
+    getConstraintSignatures,
+    type Pageable,
+  } from '$lib/sdk/fetch-client';
+  import type { DataItem } from '$lib/utils/resources';
+  import { error } from '@sveltejs/kit';
+  import { ChevronLeft } from 'lucide-svelte';
+
+  let reloadTable = {};
+
+  const getConstraints = async () => {
+    return {
+      constraints: await getConstraintSignatures({ page: 0, size: 50 }).then(({ content }) => content),
+    };
+  };
+
+  async function getInstancesPage(pageIndex: number, toSort: string, filter: string, constraintSignatureId?: string) {
+    let pageable: Pageable = { page: pageIndex, size: 5, sort: [toSort] };
+    if (!constraintSignatureId) {
+      throw error(400, { message: 'Could not fetch page' });
+    }
+    try {
+      let result = await getConstraintInstances(constraintSignatureId, pageable, { argument: filter });
+      let dataItems: DataItem[] = result.constraintInstances.map((instance) => {
+        let item: DataItem = { id: instance.id };
+        for (let i = 0; i < instance.arguments.length; i++) {
+          item[`name${i}`] = String(
```

Comment:
I don't understand what's happening here. `item` is an object with an `id` field, nothing else. And you change some random other field in there? At this point item isn't a `DateItem` anymore, is it?

Reply:
`item[`name${i}`] = ` is a way to add a new key, value pair to an object.
This is needed, because the amount of display names for the parameters are unknown at compile time, resulting in the need for a for loop to add the required key, value pairs.
`name{i}` is just a placeholder to differentiate the display names.
`DataItem` only has an id field the rest of the fields can be created by demand to fit the use case.